modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleService.java

```diff
@@ -310,7 +333,267 @@ void run(ClusterState state) {
                     e
                 );
             }
+
+            try {
+                indicesToExcludeForRemainingRun.addAll(maybeExecuteDownsampling(
+                    state,
+                    dataStream,
+                    getTargetIndices(dataStream, indicesToExcludeForRemainingRun, state.metadata()::index)
+                ));
+            } catch (Exception e) {
+                logger.error(
+                    () -> String.format(
+                        Locale.ROOT,
+                        "Data stream lifecycle failed to execute downsampling for data stream [%s]",
+                        dataStream.getName()
+                    ),
+                    e
+                );
+            }
+
+            affectedIndices += indicesToExcludeForRemainingRun.size();
+            affectedDataStreams++;
+        }
+        logger.trace(
+            "Data stream lifecycle service performed operations on [{}] indices, part of [{}] data streams",
```

Comment:
I'm thinking more about this. We unconditionally add the write index to the list, regardless of whether it was rolled over or not, which means that we'll always have at least 1 operation "performed" even if nothing happens. Should we add a way to decrease this if nothing happened to the write index, or keep it (potentially off-by-one), or remove the logging?

Reply:
>which means that we'll always have at least 1 operation "performed" even if nothing happens

I believe this is correct though? We issued a `rollover` request so Data Stream Lifecycle has performed something.

Reply:
True, though the rollover could have not actually succeeded if no conditions were met. It sounds like it's fine to keep this as-is then.